kubernetes-model-generator/kubernetes-model-core/src/main/java/io/fabric8/kubernetes/api/model/Quantity.java

```diff
@@ -249,17 +260,47 @@ public static Quantity parse(String quantityAsString) {
     if (quantityAsString == null || quantityAsString.isEmpty()) {
       throw new IllegalArgumentException("Invalid quantity string format passed.");
     }
-    String[] quantityComponents = quantityAsString.split("[eEinumkKMGTP]+");
-    String amountStr = quantityComponents[0];
-    String formatStr = quantityAsString.substring(quantityComponents[0].length());
+
+    int unitIndex = indexOfUnit(quantityAsString);
+    String amountStr = quantityAsString.substring(0, unitIndex);
+    String formatStr = quantityAsString.substring(unitIndex);
     // For cases like 4e9 or 129e-6, formatStr would be e9 and e-6 respectively
     // we need to check whether this is valid too. It must not end with character.
-    if (formatStr.matches(AT_LEAST_ONE_DIGIT_REGEX) && Character.isAlphabetic(formatStr.charAt(formatStr.length() - 1))) {
+    if (containsAtLeastOneDigit(formatStr) && Character.isAlphabetic(formatStr.charAt(formatStr.length() - 1))) {
       throw new IllegalArgumentException("Invalid quantity string format passed");
     }
     return new Quantity(amountStr, formatStr);
   }
 
+  /**
+   * @param quantityAsString quantity as a string
+   * @return the first index containing a unit character, or the length of the string if no element provided
+   */
+  static int indexOfUnit(String quantityAsString) {
+    for (int i = 0; i < quantityAsString.length(); i++) {
+      char ch = quantityAsString.charAt(i);
+      switch (ch) {
+        case 'e':
+        case 'E':
+        case 'i':
+        case 'n':
+        case 'u':
+        case 'm':
+        case 'k':
+        case 'K':
+        case 'M':
+        case 'G':
+        case 'T':
+        case 'P':
+          return i;
+        default:
+          //noinspection UnnecessaryContinue - satisfy Sonar
+          continue;
```

Comment:
Without this `default` case, [Sonar flags this as a critical `Add a default case to this switch.`](https://sonarcloud.io/project/issues?resolved=false&types=CODE_SMELL&pullRequest=4992&id=fabric8io_kubernetes-client&open=AYcPRu7YtWsAr5uOwQsf) , though with the `default` case of `continue` IntelliJ then flags it as unnecessary. Defer to your preferences if you'd like an empty `default` case or if the comment suppressing IntelliJ warning is ok with you all.

Reply:
Sonar false positives can be disabled by using a `@SuppressWarnings` annotation followed by the rule id e.g. `@SuppressWarnings("java:S107")`